tests/stage/test_stage.py

```diff
@@ -737,6 +737,46 @@ def test_execute(
     assert result.output == snapshot
 
 
+@pytest.mark.parametrize(
+    "stage_path, expected_files",
+    [
+        ("@~", ["@~/s1.sql", "@~/a/s3.sql", "@~/a/b/s4.sql"]),
+        ("@~/s1.sql", ["@~/s1.sql"]),
+        ("@~/a", ["@~/a/s3.sql", "@~/a/b/s4.sql"]),
+        ("@~/a/s3.sql", ["@~/a/s3.sql"]),
+        ("@~/a/b", ["@~/a/b/s4.sql"]),
+    ],
+)
+@mock.patch(f"{STAGE_MANAGER}._execute_query")
+def test_execute_from_user_stage(
```

Comment:
Let's add also an integration tests for this type of stage

Reply:
Added